bee-network/src/peer/info.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::Multiaddr;
+
+/// Additional information about a peer.
+#[derive(Clone, Debug)]
+pub struct PeerInfo {
+    /// The peer's address.
+    pub address: Multiaddr,
+    /// The peer's alias.
+    pub alias: String,
+    /// The type of relation we have with this peer.
+    pub relation: PeerRelation,
+}
+
+/// Describes the relation with a peer.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum PeerRelation {
+    /// Represents a known peer.
+    Known,
+    /// Represents an unknown peer.
+    Unknown,
+}
+
+impl PeerRelation {
+    /// Returns whether the peer is known.
+    pub fn is_known(&self) -> bool {
+        self.eq(&Self::Known)
+    }
+
+    /// Returns whether the peer is unknown.
+    pub fn is_unknown(&self) -> bool {
+        self.eq(&Self::Unknown)
```

Comment:
instead of calling `self.eq(&Self::Unknown)`, what about calling `!is_known()` here? what would be best-practice?

Reply:
That would probably not play very well when (if) we introduce new enum variants

Reply:
That's right. It looks like we will be having more enum variants in the future (`Discovered` possibly). Another good practice and alternative here is to use the `matches!` macro.

Reply:
`matches!`-gang assemble